src/dotnet/Telemetry.cs

```diff
@@ -19,6 +20,8 @@ public class Telemetry : ITelemetry
         private Dictionary<string, double> _commonMeasurements = null;
         private Task _trackEventTask = null;
 
+        internal static readonly string TelemetrySentinel  = Path.Combine(ApplicationEnvironment.ApplicationBasePath, $"{Product.Version}.dotnetTelemetry");
```

Comment:
Okay with the the reordering change, and a user-specific TelemetrySentinel, I think it's fine if they're separate